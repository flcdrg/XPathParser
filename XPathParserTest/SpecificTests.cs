using System.Xml.Linq;
using ApprovalTests;
using ApprovalTests.Reporters;
using CodePlex.XPathParser;
using NUnit.Framework;

namespace XPathParserTest
{
    [TestFixture]
    [UseReporter(typeof(BeyondCompareReporter))]
    public class SpecificTests
    {
        [Test]
        public void Functions()
        {
            var input = "/para[substring(\"12345\", 0, 3) = \"1234\"]";
            var actual = new XPathParser<XElement>().Parse(input, new XPathTreeBuilder());
            
            Approvals.VerifyXml(actual.ToString());
        }
    }
}