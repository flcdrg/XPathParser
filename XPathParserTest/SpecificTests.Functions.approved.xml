<step>
  <Root nodeTyepe="All" prefix="(null)" name="(null)" />
  <predicate reverse="false">
    <Child nodeTyepe="Element" prefix="" name="para" />
    <Eq>
      <function prefix="" name="substring">
        <string value="12345" />
        <number value="0" />
        <number value="3" />
      </function>
      <string value="1234" />
    </Eq>
  </predicate>
</step>